pilot/pkg/serviceregistry/kube/controller/ambient/workloads_test.go

```diff
@@ -59,6 +59,35 @@ func TestPodWorkloads(t *testing.T) {
 			},
 			result: nil,
 		},
+		{
+			name:   "simple pod not running but have podIP",
+			inputs: []any{},
+			pod: &v1.Pod{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "name",
+					Namespace: "ns",
+				},
+				Spec: v1.PodSpec{},
+				Status: v1.PodStatus{
+					Phase: v1.PodPending,
+					PodIP: "1.2.3.4",
+				},
+			},
+			result: &workloadapi.Workload{
```

Comment:
This should be fine.

Pending but with IP == CNI has had a shot already, so we are waiting on processes in the pod to spawn.

Pending with no IP == we can't guarantee CNI has had a shot already.

What we are worried about in ambient is making sure the pod netns has actually been created before we try to jump into it.

"Pod got an IP" is a good enough proxy for that, I think.

Reply:
I think we basically want: pod scheduled to node, assigned IP, and its not terminated.

Which we could do with: `ip != "" && nodeName != "" && phase in [pending, running]` or equivalent through conditions

Reply:
it's not the end of the world if it's terminated, but checking would avoid some extra churn/excess failure messages.

Reply:
```yaml
...
  status
    phase: Pending  Or Running
    podIP: 10.0.174.216
    podIPs:
    - ip: 10.0.174.216
 ```
 
 If  pod status is Pending (not terminated or failed or completed or unknow) but have been assigned IP  which mean that  pod have be scheduled to a node and is initialing.  
 
 There were related filters before, and I added the podPending judgment, which I think is consistent with what we're describing above.  This is the complete code for this logic after the merge，

```go
func (a *index) podWorkloadBuilder(
	MeshConfig krt.Singleton[MeshConfig],
	AuthorizationPolicies krt.Collection[model.WorkloadAuthorization],
	PeerAuths krt.Collection[*securityclient.PeerAuthentication],
	Waypoints krt.Collection[Waypoint],
	WorkloadServices krt.Collection[model.ServiceInfo],
) func(ctx krt.HandlerContext, p *v1.Pod) *model.WorkloadInfo {
	return func(ctx krt.HandlerContext, p *v1.Pod) *model.WorkloadInfo {
		if (!IsPodRunning(p) && !IsPodPending(p)) || p.Spec.HostNetwork {
			return nil
		}
		podIP, err := netip.ParseAddr(p.Status.PodIP)
		if err != nil {
			// Is this possible? Probably not in typical case, but anyone could put garbage there.
			return nil
		}
```

Reply:
> it's not the end of the world if it's terminated, but checking would avoid some extra churn/excess failure messages - yeah `PodReadyToStartContainers` is ideal, in contexts where it's available.

Istio now support kubernetes 1.26, 1.27, 1.28, 1.29 (https://istio.io/latest/docs/releases/supported-releases/), 

but  `PodReadyToStartContainers` was only [made default in kubernetes 1.29](https://kubernetes.io/blog/2023/12/19/pod-ready-to-start-containers-condition-now-in-beta/), which can lead to unfriendly support for other lower kubernetes versions.

Reply:
If we are going to use a custom heuristic here (Pending + some_other_conditions) then it would be good to add a comment explaining that in detail to this spot so we don't forget what and why.

We should be able to replace all those checks with a single check against `PodReadyToStartContainers` eventually, but for now I'm fine with !running + is_pending + has_ip

Reply:
Ok, thanks for your reply.

I've added the relevant notes, please check it again~